### ISSUE TITLE
Request Support for Amateur Radio Equipment and Software

### ISSUE THREAD

**Issue by G8YTZ** [bitfocus/companion-module-requests#835]
The iCOM IC-9700 amd IC-7300 are very popular and very similar multi-mode Amateur Radio Transceiver. Most iCOM models use the same protocol according to the individual model's supported  features, It has an extensive set of remote commands, but some functions, for example selecting memories and bands can be intensive. The use case here would be for Bitfocus Companion together with the Stream deck to offer direct selection keys:

1. Memory Recall for specific memories
2. Band and frequency display for main and sub receivers
3. VFO A/B selection
4. Volume up/down
5. The ability to add your own functions

Connectivity supported is via USB/Serial. 

A typical command reference can be seen here: https://www.icom-france.com/uploads/files/produit/not-IC-9700_ENG_CI-V_1-en.pdf The manufacture would in no doubt be happy to support. many products already support these protocols and libraries already exist. It might be easier and more feature rich to consider interfacing with Third Party Software such as Ham Radio Deluxe. https://www.hamradiodeluxe.com/features/ as this would open an interface beyond radios, but to log books, antennas rotator control etc.

I believe this would open a new market for Streamdeck.

**Comment by josephdadams** [comment]
So, the hardware does not have a LAN port?

**Comment by G8YTZ** [comment]
The IC-9700 and IC-7610 have a wired LAN port
The IC-705 has a wireless LAN port.
The IC-9700 is the one most of interest, but the others are very similar and built on the same software platform.

**Comment by josephdadams** [comment]
I glanced at the 9700 command reference you included and it only mentions control via usb or serial. This makes it difficult for Companion to support since it is a cross platform app, we typically only interact with remote devices through the network/ethernet stack.

**Comment by G8YTZ** [comment]
No,It definitely supports LAN. It has aIt has a built in server. No doubt about that! I have other remote control software that uses the LAN interface.are you sure it’s not referring to the very similar IC 7300 (essentially the shortwave version of the 9700 that uses earlier technology) which does not support LAN? Justin. Sent from my Remington Rand UNIVACwww.g8ytz.com On 10 Aug 2022, at 01:16, Joseph Adams ***@***.***> wrote:﻿
I glanced at the 9700 command reference you included and it only mentions control via usb or serial. This makes it difficult for Companion to support since it is a cross platform app, we typically only interact with remote devices through the network/ethernet stack.
